packages/components/pagination/src/pagination.ts

```diff
@@ -247,6 +249,12 @@ export default defineComponent({
           emit('update:page-size', v)
           emit('size-change', v)
         }
+        if (
+          !isAbsent(props.total) &&
+          currentPageBridge.value <= Math.ceil(props.total / v)
+        ) {
```

Comment:
It seems that there are still some errors，change the sizes, `@change` not triggered
[example](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPGVsLXBhZ2luYXRpb24gdi1tb2RlbDpjdXJyZW50LXBhZ2U9XCJjdXJyZW50UGFnZTJcIiB2LW1vZGVsOnBhZ2Utc2l6ZT1cInBhZ2VTaXplMlwiIDpwYWdlLXNpemVzPVwiWzEwMCwgMjAwLCAzMDAsIDQwMF1cIlxuICAgIGxheW91dD1cInRvdGFsLCBzaXplcywgcHJldiwgcGFnZXIsIG5leHQsIGp1bXBlclwiIDpwYWdlLWNvdW50PVwicGFnZUNvdW50XCIgQHNpemUtY2hhbmdlPVwic2l6ZUNoYW5nZVwiXG4gICAgQGN1cnJlbnQtY2hhbmdlPVwiY3VycmVudENoYW5nZVwiIEBjaGFuZ2U9XCJoYW5kbGVDaGFuZ2VcIiAvPlxuPC90ZW1wbGF0ZT5cbjxzY3JpcHQgc2V0dXAgbGFuZz1cInRzXCI+XG5pbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5cbmNvbnN0IGN1cnJlbnRQYWdlMiA9IHJlZigyKVxuY29uc3QgcGFnZVNpemUyID0gcmVmKDEwMClcbmNvbnN0IHBhZ2VDb3VudCA9IHJlZig3KVxuXG5jb25zdCBoYW5kbGVDaGFuZ2UgPSAoY3VycmVudFBhZ2U6IG51bWJlciwgcGFnZVNpemU6IG51bWJlcikgPT4ge1xuICBjb25zb2xlLmxvZyhgY3VycmVudDogJHtjdXJyZW50UGFnZX07IHNpemU6ICR7cGFnZVNpemV9YClcbn1cbmNvbnN0IGN1cnJlbnRDaGFuZ2UgPSAoY3VycmVudFBhZ2U6IG51bWJlcikgPT4ge1xuICBjb25zb2xlLmxvZyhgY3VycmVudDogJHtjdXJyZW50UGFnZX07YClcbn1cbmNvbnN0IHNpemVDaGFuZ2UgPSAocGFnZVNpemU6IG51bWJlcikgPT4ge1xuICBjb25zb2xlLmxvZygnc2l6ZScsIHBhZ2VTaXplKVxufVxuPC9zY3JpcHQ+XG4iLCJzcmMvUGxheWdyb3VuZE1haW4udnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCBBcHAgZnJvbSAnLi9BcHAudnVlJ1xuaW1wb3J0IHsgc2V0dXBFbGVtZW50UGx1cyB9IGZyb20gJy4vZWxlbWVudC1wbHVzLmpzJ1xuc2V0dXBFbGVtZW50UGx1cygpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8QXBwIC8+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwiZWxlbWVudC1wbHVzXCI6IFwiaHR0cHM6Ly9wcmV2aWV3LTE0ODI4LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguZnVsbC5taW4ubWpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXMvXCI6IFwidW5zdXBwb3J0ZWRcIlxuICB9XG59IiwidHNjb25maWcuanNvbiI6IntcbiAgXCJjb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwibW9kdWxlXCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVSZXNvbHV0aW9uXCI6IFwiQnVuZGxlclwiLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdLFxuICAgIFwiYWxsb3dJbXBvcnRpbmdUc0V4dGVuc2lvbnNcIjogdHJ1ZSxcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZVxuICB9LFxuICBcInZ1ZUNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogMy4zXG4gIH1cbn1cbiIsInNyYy9lbGVtZW50LXBsdXMuanMiOiJpbXBvcnQgeyBnZXRDdXJyZW50SW5zdGFuY2UgfSBmcm9tICd2dWUnXG5pbXBvcnQgRWxlbWVudFBsdXMgZnJvbSAnZWxlbWVudC1wbHVzJ1xuXG5sZXQgaW5zdGFsbGVkID0gZmFsc2VcbmF3YWl0IGxvYWRTdHlsZSgpXG5cbmV4cG9ydCBmdW5jdGlvbiBzZXR1cEVsZW1lbnRQbHVzKCkge1xuICBpZiAoaW5zdGFsbGVkKSByZXR1cm5cbiAgY29uc3QgaW5zdGFuY2UgPSBnZXRDdXJyZW50SW5zdGFuY2UoKVxuICBpbnN0YW5jZS5hcHBDb250ZXh0LmFwcC51c2UoRWxlbWVudFBsdXMpXG4gIGluc3RhbGxlZCA9IHRydWVcbn1cblxuZXhwb3J0IGZ1bmN0aW9uIGxvYWRTdHlsZSgpIHtcbiAgcmV0dXJuIG5ldyBQcm9taXNlKChyZXNvbHZlLCByZWplY3QpID0+IHtcbiAgICBjb25zdCBsaW5rID0gZG9jdW1lbnQuY3JlYXRlRWxlbWVudCgnbGluaycpXG4gICAgbGluay5yZWwgPSAnc3R5bGVzaGVldCdcbiAgICBsaW5rLmhyZWYgPSAnaHR0cHM6Ly9wcmV2aWV3LTE0ODI4LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguY3NzJ1xuICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignbG9hZCcsIHJlc29sdmUpXG4gICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdlcnJvcicsIHJlamVjdClcbiAgICBkb2N1bWVudC5ib2R5LmFwcGVuZChsaW5rKVxuICB9KVxufSIsIl9vIjp7InNob3dIaWRkZW4iOnRydWUsInN0eWxlU291cmNlIjoiaHR0cHM6Ly9wcmV2aWV3LTE0ODI4LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguY3NzIn19)

Reply:
The props.total is undefined. I'll take care of it.    >_<

Reply:
`if (
          currentPageBridge.value <=
          Math.ceil((props.total ?? Number.POSITIVE_INFINITY) / v)
        ) {
          emit('change', currentPageBridge.value, v)
        }`
I'm going to change it this way. @FrontEndDog

Reply:
I'll finish the modification and push the code tonight. @FrontEndDog

Reply:
You can take a look at my approach, Can it solve your problem.

In the judgment statement you mentioned, if `props.total===0`, there may be unexpected performance.

When I was writing this paragraph, btea proposed a better solution.